Src/IronPython/Runtime/Operations/IntOps.cs

```diff
@@ -422,14 +419,155 @@ public static object from_bytes(CodeContext context, PythonType type, object byt
             } else {
                 digits = "10000000000000000000000000000000";
             }
-            
+
             if (includeType) {
                 digits = "0b" + digits;
             }
             return digits;
         }
 
         #endregion
+
+        #region Mimic BigInteger members
+        // Ideally only on instances
+
+        #region Properties
+
+        [SpecialName, PropertyMethod, PythonHidden]
+        public static bool GetIsEven(int self) => (self & 1) == 0;
+
+        [SpecialName, PropertyMethod, PythonHidden]
+        public static bool GetIsOne(int self) => self == 1;
+
+        [SpecialName, PropertyMethod, PythonHidden]
+        public static bool GetIsPowerOfTwo(int self) => self > 0 && (self & (self - 1)) == 0;
+
+        [SpecialName, PropertyMethod, PythonHidden]
+        public static bool GetIsZero(int self) => self == 0;
+
+        [SpecialName, PropertyMethod, PythonHidden]
+        public static int GetSign(int self) => self == 0 ? 0 : self > 0 ? 1 : -1;
+
+        [PythonHidden]
+        public static object Zero => ScriptingRuntimeHelpers.Int32ToObject(0);
+
+        [PythonHidden]
+        public static object One => ScriptingRuntimeHelpers.Int32ToObject(1);
+
+        [PythonHidden]
+        public static object MinusOne => ScriptingRuntimeHelpers.Int32ToObject(-1);
+
+        #endregion
+
+        #region Methods
+
+        [PythonHidden]
+        public static byte[] ToByteArray(int self) => new BigInteger(self).ToByteArray();
+
+#if NETCOREAPP
+        [PythonHidden]
+        public static byte[] ToByteArray(int self, bool isUnsigned = false, bool isBigEndian = false) => new BigInteger(self).ToByteArray(isUnsigned, isBigEndian);
+
+        [PythonHidden]
+        public static int GetByteCount(int self, bool isUnsigned = false) => new BigInteger(self).GetByteCount(isUnsigned);
+
+        [PythonHidden]
+        public static bool TryWriteBytes(int self, Span<byte> destination, out int bytesWritten, bool isUnsigned = false, bool isBigEndian = false)
+            => new BigInteger(self).TryWriteBytes(destination, out bytesWritten, isUnsigned, isBigEndian);
+
+#elif NETSTANDARD
```

Comment:
These feel a bit icky since depending on what runtime is using the .NET Standard assemblies you will get different methods. For example, if the .NET Standard assemblies are used with .NET Framework you'll might get extra methods on `Int32` based `int` which won't appear when backed by `BigInteger`.

Anyway, I don't have a good solution to propose right now so this is probably fine.

Reply:
Indeed, but I too have no better ideas. However I am not too concerned about this case: from the perspective of the Python type system consistency it is totally acceptable for some instances to have some additional attributes that other instances or their type don't have.

On the other hand, it is not OK if a type has attributes which are missing on (some) instances (without any explicit deletion, that is). Unfortunately, this is now the case on Mono. After spending an hour or so trying to fix it, I decided to simply fix the tests on Mono and will submit an issue report once this PR is merged. Maybe it will be picked up at some point in the future (or Mono becomes obsolete by .NET vNext). My enthusiasm to work around Mono quirks is quite limited, and in any case, this particular issue is not on the top of the list.

Reply:
> I decided to simply fix the tests on Mono and will submit an issue report once this PR is merged.

Fine by me. Mono is less relevant these days with .NET having good cross-platform support.